### PR TITLE
[UI/UX] Search (almost) Everything

### LINKED ISSUE #14348: [UI] Global search is visually destroyed
### Mautic Series

5.2.x series

### Mautic installed version

5.x

### Way of installing

I downloaded a release from https://www.mautic.org/mautic-releases

### PHP version

Not relevant

### What browsers are you seeing the problem on?

Not relevant

### What happened?

Changes made to the interface since 5.1 have been destroying the general appearance of the global search.

image

The fix involves redesigning the global search template.

### How can we reproduce this issue?

1. Create some contacts
2. Click on the global search on top navbar
3. Type the name of a contact
4. See

### Relevant log output

_No response_

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

### PR DESCRIPTION
**Pull request by andersonjeccel**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🟢 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #14348 

## Description

this pr fixes mentioned issue

improves global search interface:
- adds available/active/sending, pending, expired and deactivated/unavailable statuses
- adds visual hierarchy to info
- adds icons to aid in understanding stats like read count
- shows if a searched asset is an image, video, powerpoint etc
- becomes beautiful
- makes me want to start using it. do you?
- expands the publishbadge template to support a simplified mode, where it gets smaller and no animation

| Before                                 | After
| -------------------------------------- | ---
|    image                                     | image

---
### 📋 Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
1.1. Have asset, contact, user and form created (at least)
2. Click on the global search icon on top navbar
3. Search for the stuff you have and see (or search `and` to see results for all existing things in your install; this is a bug from the functionality mentioned in #14350)

---

## The original wireframe had more functionalities
My first idea would also have **actions** appearing, specifically all buttons that were added using addButton on the LOCATION_PAGE_ACTIONS and links to all pages from the main sidebar and those from the admin dropdown (both registered in PHP too).

I'd need help to make it happen, as my tries were unsuccessful.

Group 16

### CODE REVIEW

**Comment by andersonjeccel** [comment]
@escopecz this is a fix but also an improvement

there is no way to release the bug fix without the improvement

can this PR go to the 5.2.1 release?

**Comment by escopecz** [comment]
I would let it go to the 5.2.1 release as a bug fix. CC-ing @RCheesley and @jos0405 in case they don't agree.

**Review by matbcvo — CHANGES_REQUESTED** [review]
The failing functional test needs to be resolved.

https://github.com/mautic/mautic/blob/45e32dfaffcf03bf2d99500be19232d21dc0fb06/app/bundles/CoreBundle/Tests/Functional/Controller/AjaxControllerTest.php#L65

Replace:
```php
Assert::assertSame("<div class=\"panel-group\" id=\"globalSearchPanel\"></div>\n", $content['newContent']);
```
with:
```php
Assert::assertSame("<div class=\"pa-sm\" id=\"globalSearchPanel\"></div>\n", $content['newContent']);
```
to align with your changes.

Then run the code style fixer with:
```bash
ddev composer fixcs -- app/bundles/CoreBundle/Tests/Functional/Controller/AjaxControllerTest.php
```

I’m fine with this PR being merged into 5.2.1 as a bug fix.

**Comment by andersonjeccel** [comment]
@matbcvo thank you for pointing the fix :)

**Comment by matbcvo** [comment]
My bad, the previously suggested fix was incorrect. I overlooked the Twig file and didn’t realize it returns additional content when there are no results. I’ve fixed the functional test to use a regular expression, as an exact match isn’t feasible due to the presence of random text.

**Review by matbcvo — APPROVED** [review]

File: app/bundles/CoreBundle/Resources/views/Default/navbar.html.twig [hunk 1]
```diff
@@ -53,8 +53,24 @@
...
+            icon_only: true,
+            attributes: {
+            'type': 'button',
+            'data-toggle': 'modal',
+            'data-target': '#gsearchModal',
+            'class': 'search-button'
+            }
+            }
+        ]
+        } %}
```
Comment (shinde-rahul) [inline app/bundles/CoreBundle/Resources/views/Default/navbar.html.twig]:
```suggestion
        {% include '@MauticCore/Helper/button.html.twig' with {
            buttons: [
                {
                    label: 'mautic.core.search_everything',
                    tooltip_placement: 'right',
                    variant: 'ghost',
                    icon: 'ri-search-line ri-xl',
                    icon_only: true,
                    attributes: {
                    'type': 'button',
                    'data-toggle': 'modal',
                    'data-target': '#gsearchModal',
                    'class': 'search-button'
                }
                }
            ]
        } %}
```

Minor Indentation things.

File: app/bundles/CoreBundle/Resources/views/GlobalSearch/results.html.twig [hunk 2]
```diff
@@ -1,25 +1,53 @@
...
-                    </li>
-                  {% endif %}
-                {% endfor %}
-            </ul>
-        </div>
-    </div>
-{%- endfor %}
-</div>
+    {% endif %}
+</div>
```
Comment (shinde-rahul) [inline app/bundles/CoreBundle/Resources/views/GlobalSearch/results.html.twig]:
```suggestion
</div>

```

File: app/bundles/CoreBundle/Resources/views/GlobalSearch/globalsearch.html.twig [hunk 3]
```diff
@@ -1,36 +1,46 @@
...
+
+					{{ include('@MauticCore/GlobalSearch/results.html.twig', {results: results}) }}
+				</div>
+			</div>
+		</div>
+		<!-- end results -->
+
+	</div>
+
+</div>
```
Comment (shinde-rahul) [inline app/bundles/CoreBundle/Resources/views/GlobalSearch/globalsearch.html.twig]:
```suggestion
</div>

```

File: app/bundles/CoreBundle/Resources/views/GlobalSearch/globalsearch.html.twig [hunk 4]
```diff
@@ -1,36 +1,46 @@
...
+			</div>
+
+
+		</div>
+		<!-- end modal content -->
+		<!-- start results -->
+		<div class="modal-content gsearch--results hide mt-xs">
+			<div class="pt-0 pb-xs pl-0 pr-0">
+				<div class="gsearch--results-scroll scroll-content slimscroll" id="globalSearchResults">
+
```
Comment (shinde-rahul) [inline app/bundles/CoreBundle/Resources/views/GlobalSearch/globalsearch.html.twig]:
```suggestion
```

File: app/bundles/CoreBundle/Resources/views/GlobalSearch/globalsearch.html.twig [hunk 5]
```diff
@@ -1,36 +1,46 @@
...
+					{{ include('@MauticCore/GlobalSearch/results.html.twig', {results: results}) }}
+				</div>
+			</div>
+		</div>
+		<!-- end results -->
+
```
Comment (shinde-rahul) [inline app/bundles/CoreBundle/Resources/views/GlobalSearch/globalsearch.html.twig]:
```suggestion
```

File: app/bundles/CoreBundle/Resources/views/GlobalSearch/globalsearch.html.twig [hunk 6]
```diff
@@ -1,36 +1,46 @@
...
+					class="gsearch--field__input form-control search fs-16" id="globalSearchInput" name="global_search"
+					placeholder="{% trans %}mautic.core.search_everything{% endtrans %}"
+					autocomplete="false" data-toggle="livesearch" data-target="#globalSearchResults"
+					data-action="{{ path('mautic_core_ajax', {action: 'globalSearch'}) }}" data-overlay="true"
+					data-overlay-text="{% trans %}mautic.core.search.livesearch{% endtrans %}" />
+				<span class="gsearch--field-addon">
+					<kbd class="mr-2">f</kbd><kbd>/</kbd>
+				</span>
+			</div>
+
```
Comment (shinde-rahul) [inline app/bundles/CoreBundle/Resources/views/GlobalSearch/globalsearch.html.twig:18]:
```suggestion
```

File: app/bundles/CoreBundle/Resources/views/GlobalSearch/globalsearch.html.twig [hunk 7]
```diff
@@ -1,36 +1,46 @@
...
-            <div class="pt-0 pb-xs pl-0 pr-0">
-                <div class="scroll-content slimscroll" style="height:250px;" id="globalSearchResults">
-                    {{ include('@MauticCore/GlobalSearch/results.html.twig', {results: results}) }}
-                </div>
-            </div>
-        </div>
-    </div>
+	<div class="dropdown-menu">
+
+	</div>
```
Comment (shinde-rahul) [inline app/bundles/CoreBundle/Resources/views/GlobalSearch/globalsearch.html.twig]: Do you know if we need this? If so, can we add some comments to the empty div to clarify the purpose?

File: app/bundles/CoreBundle/Resources/views/GlobalSearch/globalsearch.html.twig [hunk 8]
```diff
@@ -1,36 +1,46 @@
...
-            autocomplete="false"
-            data-toggle="livesearch"
-            data-target="#globalSearchResults"
-            data-action="{{ path('mautic_core_ajax', {action: 'globalSearch'}) }}"
-            data-overlay="true"
-            data-overlay-text="{% trans %}mautic.core.search.livesearch{% endtrans %}" />
-    </div>
+	<div class="search-container" id="globalSearchContainer">
+
+	</div>
```
Comment (shinde-rahul) [inline app/bundles/CoreBundle/Resources/views/GlobalSearch/globalsearch.html.twig]: Do you know if we need this? If so, can we add some comments to the empty div to clarify the purpose?

**Review by shinde-rahul — CHANGES_REQUESTED** [review]
Thanks @andersonjeccel!!!

I have added some minor style-related observations with suggestions and a couple of questions.

File: app/bundles/CoreBundle/Resources/views/Default/navbar.html.twig (hunk 1 above)
Comment (andersonjeccel) [reply app/bundles/CoreBundle/Resources/views/Default/navbar.html.twig]: Do you use a specific formatter for Twig in VS Code?

These weird indentations happen every time when using the normal HTML formatter :/

File: app/bundles/CoreBundle/Resources/views/GlobalSearch/globalsearch.html.twig (hunk 7 above)
Comment (andersonjeccel) [reply app/bundles/CoreBundle/Resources/views/GlobalSearch/globalsearch.html.twig]: Definitely forgot to remove

File: app/bundles/CoreBundle/Resources/views/Default/navbar.html.twig (hunk 1 above)
Comment (shinde-rahul) [reply app/bundles/CoreBundle/Resources/views/Default/navbar.html.twig]: I am using PHPStrom for Mautic. I'm not sure about the VS Code formatter.

**Review by shinde-rahul — APPROVED** [review]
The code changes LGTM!!!

Thanks @andersonjeccel!!!

**Pull request merged**